### PR TITLE
refactor(archive): move to single-export files

### PR DESCRIPTION
**Pull request by iuioiua**
This is part of #2936, which aims for leaner dependencies and a more straightforward structure by only having one export per file within the archive module.

### CODE REVIEW

**Comment by iuioiua** [comment]
Do we want `_common.ts` and `_test_common.ts` or `_util.ts` and `_test_util.ts`? I'm impartial.

File: archive/mod.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+/**
+ * Provides a `Tar` and `Untar` classes for compressing and decompressing
+ * arbitrary data.
+ *
+ * @module
+ */
+/** @deprecated (will be removed after 0.169.0) TODO: export all once overlapping objects/types are removed */
+export { Tar, type TarData, type TarDataWithSource } from "./tar.ts";
+export * from "./untar.ts";
```
Comment (kt3k) [inline archive/mod.ts:40]: `mod.ts` doesn't exist now. Does it make sense to add these deprecated versions here?

Comment (iuioiua) [reply archive/mod.ts:40]: Previously, `tar.ts` acted as the module entry point. Now that `Tar` and `Untar` classes have been separated, `mod.ts` acts as the entry point for both classes.

Comment (iuioiua) [reply archive/mod.ts:40]: It'll also provide a module description where [one currently doesn't exist for `std/archive`](https://deno.land/std?doc).

**Comment by kt3k** [comment]
> Do we want _common.ts and _test_common.ts or _util.ts and _test_util.ts? I'm impartial.

hmm both seem having examples:

```
crypto/_util.ts
fs/_util.ts
media_types/_util.ts
node/path/_util.ts
path/_util.ts

vs

node/_fs/_fs_common.ts
node/_http_common.ts
node/_tls_common.ts
streams/_common.ts
streams/_test_common.ts
uuid/_common.ts
```

I'm ok with either way.

**Comment by iuioiua** [comment]
Ok, I'll leave it as is then.

**Review by kt3k — APPROVED** [review]
LGTM

**Pull request merged**